schematic_api/api/routes.py

```diff
@@ -206,130 +220,193 @@ def get_temp_jsonld(schema_url):
-        if output_format == "excel":
+        if self.output_format == "excel":
             dir_name = os.path.dirname(result)
             file_name = os.path.basename(result)
             mimetype='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
             return send_from_directory(directory=dir_name, path=file_name, as_attachment=True, mimetype=mimetype, max_age=0)
-               
+                
         return result
 
-    # Gather all returned result urls
-    all_results = []
-    if data_type[0] == 'all manifests':
-        sg = SchemaGenerator(path_to_json_ld=jsonld)
-        component_digraph = sg.se.get_digraph_by_edge_type('requiresComponent')
-        components = component_digraph.nodes()
-        for component in components:
-            if title:
-                t = f'{title}.{component}.manifest'
-            else: 
-                t = f'Example.{component}.manifest'
-            if output_format != "excel":
-                result = create_single_manifest(data_type=component, output_format=output_format, title=t, access_token=access_token)
-                all_results.append(result)
-            else: 
-                app.logger.error('Currently we do not support returning multiple files as Excel format at once. Please choose a different output format. ')
-    else:
-        for i, dt in enumerate(data_type):
-            if not title: 
-                t = f'Example.{dt}.manifest'
-            else: 
-                if len(data_type) > 1:
-                    t = f'{title}.{dt}.manifest'
-                else: 
-                    t = title
-            if dataset_ids:
-                # if a dataset_id is provided add this to the function call.
-                result = create_single_manifest(data_type=dt, dataset_id=dataset_ids[i], output_format=output_format, title=t, access_token=access_token)
-            else:
-                result = create_single_manifest(data_type=dt, output_format=output_format, title=t, access_token=access_token)
-
-            # if output is pandas dataframe or google sheet url
-            if isinstance(result, str) or isinstance(result, pd.DataFrame):
-                all_results.append(result)
-            else: 
-                if len(data_type) > 1:
-                    app.logger.warning(f'Currently we do not support returning multiple files as Excel format at once. Only {t} would get returned. ')
-                return result
-
-    return all_results
+    @staticmethod
+    def get_manifests_route(schema_url: str, data_type:List[str], use_annotations: Optional[str]=None, dataset_id: Optional[List[str]]=None, title: Optional[str]=None, asset_view: Optional[str]=None, output_format: Optional[str]="google_sheet") -> str|pd.DataFrame|BinaryIO:
+        """Generate a new manifest template or create an existing manifest in google sheet/excel/dataframe format. 
+
+        Args:
+            schema_url (str): data model in json ld format.
```

Comment:
There's a [Python package](https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/d6e5782e6849bd47932f1a18201c3abb4eb79ac5/setup.cfg#L96) that will use type hints to document arguments. This way, you don't need to include this redundant information in the docstrings, which will likely become inconsistent over time.

Reply:
Thank you Bruno! Let me create a separate PR for this.